model_compression_toolkit/qat/keras/quantizer/README.md

```diff
@@ -0,0 +1,73 @@
+## Introduction
+
+The `BaseKerasTrainableQuantizer` class is a base class for Keras quantizers that are used for training only. It is a subclass of `BaseTrainableQuantizer` and is designed to be used as a base class for creating custom quantization methods for Keras models.
+
+## Installation
+
+To use the `BaseKerasTrainableQuantizer` class, you will need to have Tensorflow installed. If Tensorflow is not installed, an exception will be raised.
+
+Once you have Tensorflow installed, you can use the `BaseKerasTrainableQuantizer` class by importing it from the package where it is located.
+
+## Usage
+
+The `BaseKerasTrainableQuantizer` class takes single argument during initialization:
+
+- `quantization_config`: quantization configuration object which can be `NodeWeightsQuantizationConfig` or `NodeActivationQuantizationConfig` for weights or activations respectively.
+
+Once you have instantiated the class, you can use the `__call__` method to quantize the given inputs while training using the quantizer parameters. 
+The method returns the quantized tensor. You must implement the abstract method `__call__` in your subclass which inherits `BaseKerasTrainableQuantizer`.
+
+In order to export the quantizer for deployment you also must to implement `convert2inferable` method. For that you need to create your Inferable Quantizer.
+
+Example for custom symmetric weights training quantizer.
+First, add enum to `TrainingMethod` in `qat/common/qat_config.py` your custom training method and set your custom quantizer to `quantization_dispatcher_builder.py` dictionary.
+```python
+class TrainingMethod(Enum):
+    STE = 0
+    MyTrainingMethod = 1
+```
+```python
+METHOD2WEIGHTQUANTIZER = { ...,
+                           TrainingMethod.MyTrainingMethod: {qi.QuantizationMethod.SYMMETRIC: MyWeightsTrainingQuantizer}}
+```
+then, create your custom quantizer:
+```python
+from model_compression_toolkit import quantizers_infrastructure as qi
+class MyWeightsTrainingQuantizer(BaseKerasTrainableQuantizer):
```

Comment:
Doesn't it have to implement get_config as well?

Reply:
no